### ISSUE TITLE
Phylogenetic Analysis Odd Result

### ISSUE THREAD

**Issue by whottel** [UPHL-BioNGS/Grandeur#143]
Hello,

I recently ran the grandeur pipeline version 3.5.20231024 and observed an odd result.
Sequence_8 is roughly 1000 SNPs different from the other sequences I included in this analysis, except for Sequence_7 that only differs by 37 SNPs. This does not make logical sense to me as Sequence_7 only differs by 9-93 SNPs from those same other sequences. Odd things like this I have seen before if there are too few core genes being considered in the comparison, but there were 3,200 core genes included in this comparison well above 1,500 as recommended in the grandeur documentation.
Any thoughts on what could be causing this result?
[grandeur_snp_matrix - Copy.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13209390/grandeur_snp_matrix.-.Copy.csv)
[grandeur_roary_metrics_mqc.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13209391/grandeur_roary_metrics_mqc.csv)

Thanks,
Wes

**Comment by erinyoung** [comment]
My guess? Roary dropped a bunch of genes of Sequence 7.

Can you try rerunning the workflow (not with resume, but starting with fasta files should be okay)? If it's roary dropping genes, then that should fix the issue.

**Comment by whottel** [comment]
I got very simialr results when re-running starting with the fastas.
[snp_matrix_fastas_rerun.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13210062/snp_matrix_fastas_rerun.csv)
[roary_metrics_mqc_fastas_rerun.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13210067/roary_metrics_mqc_fastas_rerun.csv)

**Comment by erinyoung** [comment]
Oh snap!

**Comment by erinyoung** [comment]
What does the sequence of Sequence 7 look like? Are there a lot of non-ambiguous bases?

Something you can attempt is to take ambiguous bases into account with `-a`. This can be added to snp-dists in Grandeur with the snp_dists_options options.

```
nextflow run UPHL-BioNGS/Grandeur <everything else you use> -resume -snp_dists_options "-c -a"
```

**Comment by whottel** [comment]
I ran grandeur again using the options you described, and certainly a very different count is produced. 
[snp_matrix - Copy.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13220365/snp_matrix.-.Copy.csv)
My interpretation here is that the relatively high number of differences when including Ns observed with Sequence_7 is an indication that including this sequence in the analysis is throwing things off even though there is a high number of core genes. I took this sequence out. And now everything seems to be more reasonable. 
[snp_matrix_removed_Sequence_7.csv](https://github.com/UPHL-BioNGS/Grandeur/files/13220630/snp_matrix_removed_Sequence_7.csv)

And for a tangent thinking about the core gene used in this analysis. Would plasmid reads and/or contaminating viral reads that happen to be present in every sequence be included in the analysis as core genes? Or would these be masked by some process?

**Comment by erinyoung** [comment]
> My interpretation here is that the relatively high number of differences when including Ns observed with Sequence_7 is an indication that including this sequence in the analysis is throwing things off even though there is a high number of core genes.

I agree with your interpretation.

> And for a tangent thinking about the core gene used in this analysis. Would plasmid reads and/or contaminating viral reads that happen to be present in every sequence be included in the analysis as core genes? Or would these be masked by some process?

If the gene is found in all samples, it gets included in the core genome. This gene could be on some sort of mobile element. 

But... in theory... if you are dealing with a recent clonal expansion or an outbreak, a lot of the mobile elements should be shared.

**Comment by whottel** [comment]
Great, thank you!

**Comment by whottel** [comment]
Thinking back on this issue, I have a couple of follow-up questions. When I ran grandeur starting from the fastas files, I did not see any N’s in the assembly for Sequence 7 (0 N’s per 100 kbp in quast). So, my question is when are these N’s getting introduced? Is it that there is enough of a gene present in Sequence 7 for it to be annotated by prokka then when roary performs it alignment those genes are included and fills in N’s (or roary I believe uses “-“)? Can a high number of N’s present in the roary core alignment be easily checked using the existing roary output files that are generated? Or is invoking ‘-snp_dists_options "-c -a"’ the simplest way to check for this?
